modules/xfeatures2d/include/opencv2/xfeatures2d.hpp

```diff
@@ -179,6 +179,55 @@ class CV_EXPORTS_W LATCH : public Feature2D
     CV_WRAP static Ptr<LATCH> create(int bytes = 32, bool rotationInvariance = true, int half_ssd_size = 3, double sigma = 2.0);
 };
 
+/*
+* BEBLID Descriptor
+*/
+
+/** @brief Class implementing BEBLID (Boosted Efficient Binary Local Image Descriptor)
+ * @cite Suarez2020BEBLID .
+
+BEBLID \cite Suarez2020BEBLID is a efficient binary descriptor learned with boosting.
+It is able to describe keypoints from any detector just by changing the scale_factor parameter.
+In several benchmarks it has proved to largely improve other binary descriptors like ORB or
+BRISK with the same efficiency. BEBLID describes using the difference of mean gray values in
+different regions of the image around the KeyPoint, the descriptor is specifically optimized for
+image matching and patch retrieval addressing the asymmetries of these problems.
+
+If you find this code useful, please add a reference to the following paper:
+<BLOCKQUOTE> Iago Suárez, Ghesn Sfeir, José M. Buenaposada, and Luis Baumela.
+BEBLID: Boosted efficient binary local image descriptor.
+Pattern Recognition Letters, 133:366–372, 2020. </BLOCKQUOTE>
+
+The descriptor was trained using 1 million of randomly sampled pairs of patches
+(20% positives and 80% negatives) from the Liberty split of the UBC datasets
+\cite winder2007learning as described in the paper @cite Suarez2020BEBLID.
+You can check in the [AKAZE example](https://raw.githubusercontent.com/opencv/opencv/master/samples/cpp/tutorial_code/features2D/AKAZE_match.cpp)
+how well BEBLID works. Detecting 10000 keypoints with ORB and describing with BEBLID obtains
+561 inliers (75%) whereas describing with ORB obtains only 493 inliers (63%).
+*/
+class CV_EXPORTS_W BEBLID : public Feature2D
+{
+public:
+    /**
+     * @brief  Descriptor number of bits, each bit is a boosting weak-learner.
+     * The user can choose between 512 or 256 bits.
+     */
+    CV_WRAP enum
```

Comment:
Add enum name like `DESCRIPTOR_SIZE`

Reply:
I have added a new enum **`BeblidSize`**. It gave me some problem with the python wrappers, so I imitated the Daisy strategy adding the enum to the files [`gen_dict.json`](https://github.com/opencv/opencv_contrib/pull/2774/commits/0f10bd3ff16c6eb285dda2dbabd36675de544d10#diff-acd9128554f80b55f5448d9e6a7b4769156f30911e06b955a012def9ea1eb9d7) and [`pyopencv_xfeatures2d.hpp`](https://github.com/opencv/opencv_contrib/pull/2774/commits/0f10bd3ff16c6eb285dda2dbabd36675de544d10#diff-ca37445c0bbe1a70f0642d66ddcbfcb5dc6faf2a7cecfe3bf1a5893e3a2d1b15). Please, let me known if this solution is valid or there is a better way.